frontend/README

```diff
@@ -1,90 +1,59 @@
-structure used to store sources and headers. This addresses three problems:
-
- 1. it's not always clear where to find a header/source file - the
-    convention varies within the compiler
- 2. the current structure is not compatible with refactoring the compiler
-    into a library and multiple tools using that library
- 3. the current structure makes namespacing harder
-
-### old compiler structure
-
-The current structure is to
- * separate out the .cpp files into subdirectories by topic (AST,
-   resolution, optimizations). These are currently stored under
-   `compiler` so, for example, we have `compiler/AST`.
- * store all include files used by multiple topics in `compiler/include`.
- * store include files used only within a single topic within that topic
-   directory. For example, we have some header files in
-   `compiler/resolution`.
-
-### dyno structure
-
-We generally try to follow the LLVM project's directory structure. The
-structure (for each LLVM project) is this:
-
- * For code that goes into the library:
-   * store .cpp files in `lib/<topic>`
-   * store C++ `.h` files forming the library API into
-     `include/<project-name>/ and C API `.h` files (if any) in
-     `include/<project-name>-c/`. Header files that are relevant to a
-     particular topic go into `include/<project-name>/<topic>` and almost
-     all header files go into these topic subdirectories. The
-     <project-name> here is something like 'chpl' for us and this just
-     makes it so that the headers included have the same path as if they
-     were installed in e.g. `/usr/include`. So you can have e.g.
-     #include "chpl/uast/BlockStmt.h".
-   * store `.h` files that are for use only within the project in
-     `lib/<topic>`
-   * For example, within the LLVM project there is
-     `include/llvm/Analysis/AliasAnalysis.h` and
-     `llvm/lib/Analysis/AliasAnalysis.cpp`.
-
- * For code that does not go into a library and only supports a tool,
-   store it in `tool/<tool-name>`. For example, there is an LLVM tool
-   called `opt` and the sources for it are stored in `tools/opt`.
-
-In particular, for our project, I would expect we end up with this structure:
-```
-  runtime/ (could also be called chpl-rt/ in the future)
-  compiler/  (could also be called chpl/ in the future)
-     include/chpl
-       uast/ (new AST headers forming library API will go here)
-       resolution/ (headers forming API for new resolver)
-       (nothing from the old compiler since it does not present a library interface)
-
-     lib/
-       uast/ (new AST source code)
-       resolution/ (resolution code)
-       ...
-
-     tools/
-        chpl/
-          include/ (current contents of compiler/include)
-          AST/ (current contents of compiler/AST)
-          ...
-          main/ (source code only used in `chpl` and not in compiler-as-a-library - could contain contents of `compiler/main` today)
-```
+This directory contains the implementation of the Chapel compiler
```

Comment:
I personally would like if this README had a header before the main text, maybe "Chapel Compiler Frontend Library" or something. This would give it an identity besides of "the README in `frontend/`" which to me seems easier to start reading.